scaleway/resource_object_bucket_lock_configuration_test.go

```diff
@@ -0,0 +1,241 @@
+package scaleway
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	sdkacctest "github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+const (
+	LockResourcePrefix   = "tf-acc-test"
+	lockResourceTestName = "scaleway_object_bucket_lock_configuration.test"
+)
+
+func TestAccObjectBucketLockConfiguration_basic(t *testing.T) {
+	if !*UpdateCassettes {
+		t.Skip("Skipping ObjectStorage test as this kind of resource can't be deleted before 24h")
+	}
+	rName := sdkacctest.RandomWithPrefix(LockResourcePrefix)
+	resourceName := lockResourceTestName
+
+	tt := NewTestTools(t)
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ErrorCheck:        ErrorCheck(t, EndpointsID),
+		ProviderFactories: tt.ProviderFactories,
+		CheckDestroy:      testAccCheckBucketLockConfigurationDestroy(tt),
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(`
+					resource "scaleway_object_bucket" "test" {
+						name = %[1]q
+						tags = {
+							TestName = "TestAccSCW_LockConfig_basic"
+						}
+
+						object_lock_enabled = true
+					}
+
+					resource "scaleway_object_bucket_acl" "test" {
+						bucket = scaleway_object_bucket.test.name
+						acl = "public-read"
+					}
+
+					resource "scaleway_object_bucket_lock_configuration" "test" {
+						bucket = scaleway_object_bucket.test.name
+						rule {
+							default_retention {
+								mode = "GOVERNANCE"
+								days = 1
+							}
+						}
+					}
+				`, rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckBucketLockConfigurationExists(tt, resourceName),
+					resource.TestCheckResourceAttrPair(resourceName, "bucket", "scaleway_object_bucket.test", "name"),
+					resource.TestCheckResourceAttr(resourceName, "rule.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.default_retention.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.default_retention.0.mode", "GOVERNANCE"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.default_retention.0.days", "1"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccObjectBucketLockConfiguration_update(t *testing.T) {
+	if !*UpdateCassettes {
+		t.Skip("Skipping ObjectStorage test as this kind of resource can't be deleted before 24h")
+	}
+	rName := sdkacctest.RandomWithPrefix(LockResourcePrefix)
+	resourceName := lockResourceTestName
+
+	tt := NewTestTools(t)
```

Comment:
Do not skip tests with cassettes and init correctly TestTools, defer Cleanup is missing